### PR TITLE
chore(telemetry): simplify otel configurations and add otel configs to app_started

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Currently otel support is configured by setting two environment variables. 

- Ensures ``OTEL_PYTHON_CONTEXT`` is set when otel is enabled and ddtrace is imported
   - Previously this envar was only set when ddtrace-run was used.
   - This simplifies otel configurations and allows us to remove ``OTEL_PYTHON_CONTEXT`` from public documentation.
- Sends remaining otel configuration, ``DD_TRACE_OTEL_ENABLED``, to metabase (via instrumentation)
- Tracks whether otel was enabled using ddtrace_run or programatically.

## Testing Strategy 

Manually verify that payloads are received and processed in metabase.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](../docs/contributing.rst#release-branch-maintenance))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](../docs/contributing.rst#release-branch-maintenance)

### CODE REVIEW

File: ddtrace/opentelemetry/__init__.py
Comment (mabdinur) [inline ddtrace/opentelemetry/__init__.py:22]:
```suggestion

    import os
```

File: ddtrace/bootstrap/sitecustomize.py
Comment (brettlangdon) [inline ddtrace/bootstrap/sitecustomize.py:262]: is this accurate? should we instead emit an env var or something from `ddtrace/commands/ddtrace_run.py` ?

since `ddtrace/auto.py` just imports `ddtrace.bootstrap.sitecustomize`

Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:262]: I don't think it matters if `ddtrace-run` is used or `ddtrace.bootstrap.sitecustomize` or `ddtrace.auto` is imported. I just want to track if this file is executed.

Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:262]: I considered naming the telemetry configuration `ddtrace_was_boostraped` but that name feels a bit cryptic.

File: tests/telemetry/test_writer.py
Comment (mabdinur) [inline tests/telemetry/test_writer.py:92]: ?
```suggestion
                "name": "ddtrace_bootstraped",
```

File: ddtrace/bootstrap/sitecustomize.py
Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:262]: 🤔 I _think_ we do want to know the difference?

since with `ddtrace-run` we are controlling imports and patching for a specific ordering, and with `import ddtrace.auto` they can still import that in an incorrect location.... ?

checking `sys.modules` could be "interesting" approach (not convinced this is a good idea, was just curious what it could look like)

```python
import sys


ddtrace_auto_used = "ddtrace.auto" in sys.modules
```

but with `ddtrace-run` we don't get `ddtrace.bootstrap.sitecustomize` since we change the path, so it gets imported as `sitecustomize`, but it could be any `sitecustomize`, not necessarily ours.


(these are random musings, not a request for a change)

Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:262]: Yup, we need to check sys.path to get if `ddtrace_run` was used.

Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:262]: At least for otel it does not matter. Both will configure the tracerprovider. But in general distinguishing between the two can help us resolve escalations.

Maybe we should send another configuration, `ddtrace_auto_used`.

File: ddtrace/internal/telemetry/writer.py
Comment (brettlangdon) [inline ddtrace/internal/telemetry/writer.py:458]: @mabdinur something @XG-xin and I were talking about for https://github.com/DataDog/dd-trace-py/pull/6148

we might want to try and store these values differently, e.g. store them globally or as a class attribute rather than writing statically into the payload like this. since we need to compare against previous values in order to track changes we don't want to have to look through the payload for the previous values.

just calling it out as something that could change on you.

Comment (mabdinur) [reply ddtrace/internal/telemetry/writer.py:458]: Ahh okay. Ya this isn't the best approach. I just wanted to close out the otel work and then refactor later. I can hold off for now.

Comment (brettlangdon) [reply ddtrace/internal/telemetry/writer.py:458]: I'll let the two of you decide how to handle. I don't see why order should matter too much, as long as you coordinate.

File: ddtrace/bootstrap/sitecustomize.py
Comment (brettlangdon) [inline ddtrace/bootstrap/sitecustomize.py:214]: does the order of this matter?

Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:214]: we would be delaying to later, no?

File: ddtrace/settings/config.py
Comment (brettlangdon) [inline ddtrace/settings/config.py:311]: could this be too late?

File: ddtrace/bootstrap/sitecustomize.py
Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:214]: Order doesn't really matter as long as this environment variable is set before the first otel span is generated.
This will actually occur earlier since the config object is one of the first things imported by ddtrace.

Comment (brettlangdon) [reply ddtrace/bootstrap/sitecustomize.py:214]: what happens if we change this env var after the first span is created?

when does OTel look at this env variable?

File: ddtrace/internal/telemetry/writer.py
Comment (mabdinur) [reply ddtrace/internal/telemetry/writer.py:458]: I'll reach out to @XG-xin. I would prefer to get this in as soon as possible.

File: ddtrace/settings/config.py
Comment (mabdinur) [reply ddtrace/settings/config.py:311]: see above

**Review by emmettbutler — APPROVED** [review]

File: ddtrace/internal/telemetry/writer.py
Comment (XG-xin) [reply ddtrace/internal/telemetry/writer.py:458]: @mabdinur I update my PR to store the configuration queue in the class attribute here:
https://github.com/DataDog/dd-trace-py/pull/6148/files#diff-0cb9315daf5ed5208de657467cc10a6b7dda616a161c9b5b1f8cff83c95894deR392
Does this look good to you?

File: ddtrace/bootstrap/sitecustomize.py
Comment (mabdinur) [reply ddtrace/bootstrap/sitecustomize.py:214]: Nope. In that scenario a noop tracer provider is set and spans are not sent. 

A warning log is generated in this scenario. It's easy to debug.

**Pull request merged**
